packages/baggage-span-processor/src/baggage-span-processor.ts

```diff
@@ -43,6 +43,19 @@ import {
  * values will appear in all outgoing HTTP headers from the application.
  */
 export class BaggageSpanProcessor implements SpanProcessor {
+  private _keyFilter: (key: string) => boolean;
+
+  /**
+   * Constructs a new BaggageSpanProcessor instance.
+   * @param keyFilter An optional function that can be used to filter keys
+   * added to the span. If the function returns true, the key will be added
+   * to the span. If the function returns false, the key will not be added
+   * to the span. The default behavior is to add all keys to the span.
+   */
+  constructor(keyFilter?: (key: string) => boolean) {
```

Comment:
perhaps instead of `keyFilter` we can call it predicate? I am thinking something like: `keyRecordingPredicate` or `baggageKeyPredicate`. The use of predicate already self explains that this is a function that receives a key to filter and returns boolean to include or filter it.